### ISSUE TITLE
Feature Request: Increase the number of keywords List/set

### ISSUE THREAD

**Issue by oirfeodent** [notepad-plus-plus/notepad-plus-plus#6020]
### Description of the Issue

The list of keyword sets currently used by NPP seems to be 7.
There are some cases where the keyword list used by some languages goes beyond 7.
It could be useful to increase the keyword list to possibly 9/10.

### Steps to Reproduce the Issue

Not sure if the code location can be mentioned here... but will give a try.
If there is need to amend this, will do later.
1. Currently [**notepad-plus-plus\PowerEditor\src\ScitillaComponent\ScintillaEditView::setLexer()**](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/92bad0a60ad606b30df9ed97aecf4ff27bb6e967/PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp#L1179) is used to set the keyword list for any language.
2. Though the array [**(*pKwArray[10])**](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/92bad0a60ad606b30df9ed97aecf4ff27bb6e967/PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp#L1183) is declared as 10, only 7 is defined to be used with [**setLexer()**](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/92bad0a60ad606b30df9ed97aecf4ff27bb6e967/PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp#L1179)
3. Due to this the languages which have more keywords beyond 7 cannot use those keywords list.

### Expected Behavior

If possible to increase the keyword list/set increase the same to the 10 available.

### Actual Behavior

Cannot use more than 7 keyword list/set.

### Debug Information

Notepad++ v7.7.1   (64-bit)
Build time : Jun 16 2019 - 21:24:47
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS : Windows 8.1 (64-bit)
Plugins : BetterMultiSelection.dll ComparePlugin.dll HTMLTag.dll JSMinNPP.dll LanguageHelp.dll LuaScript.dll mimeTools.dll NppConverter.dll NppEventExec.dll NppExec.dll NppExport.dll NppFTP.dll NPPJSONViewer.dll NppJumpList.dll NppQrCode64.dll NppRegExTractorPlugin.dll NppSaveAsAdmin.dll PreviewHTML.dll SurroundSelection.dll TagLEET.dll XMLTools.dll zoomdisabler_x64.dll

**Comment by Ekopalypse** [comment]
Interestingly the [keywordset_max](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/92bad0a60ad606b30df9ed97aecf4ff27bb6e967/scintilla/include/Scintilla.h#L1003) is set to 30.
Nevertheless, at the end it is the lexer which might not expect/read more defined keywordsets, meaning,
if the lexer only expects two keywordlists it doesn't help to define 10 which, afaik, none of the lexers use atm. BAAN using 9 seems to be the the upper limit.

**Comment by oirfeodent** [comment]
@Ekopalypse 
Right, Baan using 9 is the upper limit currently. 
I authored the Baan Lexer on Scintilla... which increased the keyword set from 2 to 9 for Baan.
I created a PR #5808 , but did not follow the protocol of starting the PR from an issue.
Hence it was rejected and I have created this issue. 

I have another PR specifically for BaanC #5807 where, it uses the maximum of [7 Keywords](https://github.com/notepad-plus-plus/notepad-plus-plus/pull/5807/files#diff-31a83cf51618451fdb517e068e9068cbR881) currently. 
If this issue gets accepted and the equivalent PR gets accepted as well... I will create another PR to use the 9 keyword sets for BaanC. 

However, since [*pKwArray[10]](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/92bad0a60ad606b30df9ed97aecf4ff27bb6e967/PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp#L1183) is declared as 10... I could do a PR for increasing it to 10 as well... 
Nevertheless, as you rightly said only 9 will be used currently and only by Baan (in that case, I could reopen the current PR #5807 as is and wait for its merge).
